### PR TITLE
Iceperf example updated - ein bisschen less advertisement

### LINKED ISSUE #2003: Extend 'iceperf' with 'WaitSet'
## Brief feature description

Currently `iceperf` supports only the polling API. In order to get an idea about the event based performance `iceperf` should be extended with the `WaitSet`

### PR DESCRIPTION
**Pull request by anderay**
## Pre-Review Checklist for the PR Author

1. [x] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [x] Tests follow the [best practice for testing][testing]
1. [x] Changelog updated [in the unreleased section][changelog] including API breaking changes
1. [x] Branch follows the naming format (`iox-123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] All touched (C/C++) source code files from `iceoryx_hoofs` are added to `./clang-tidy-diff-scans.txt`
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/concepts/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/release-notes/iceoryx-unreleased.md

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] All touched (C/C++) source code files from `iceoryx_hoofs` have been added to `./clang-tidy-diff-scans.txt`
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- Closes #2003

### CODE REVIEW

**Comment by elfenpiff** [comment]
@anderay Thanks for the contribution!

Could you please follow the contribution guidelines from here: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md

Things you still have to do:
1. Sign the Eclipse Committer Agreement
2. Create an issue what you implemented here so that other reviewers see the intention of the changes.
3. The pull request name, every comment etc. should be in the english language
4. The PR should have the prefix `iox-???` where ??? is the issue number (the issue you created in 2.)
5. Every commit requires the prefix `iox-#???` (do not forget the `#` here)

Please remove all the commented out code.

As soon this is realized I am happy to review the PR.

**Comment by elBoberido** [comment]
@anderay please also run `clang-format` on your code.

I skimmed over the PR and it is great to soon have a performance test with the `WaitSet`. I had it since ages on my to-do list so I'm happy to see this PR.

File: iceoryx_examples/iceperf/iceoryx.cpp
Comment (elBoberido) [inline iceoryx_examples/iceperf/iceoryx.cpp:29]: I had to think a bit what `eventName` means. I guess this comes from using `eventId` when we talk about this part of the `ServiceDescription`

File: iceoryx_examples/iceperf/iceoryx_wait.cpp
Comment (elBoberido) [inline iceoryx_examples/iceperf/iceoryx_wait.cpp:1]: Please also add your copyright at the top of the file

File: iceoryx_examples/iceperf/iceperf_leader.cpp
Comment (elBoberido) [inline iceoryx_examples/iceperf/iceperf_leader.cpp:101]: I have to run and check the output. Currently I somewhat fear that for large payloads it isn't anymore easily distinguishable how large they actually are.

One idea would be to have an output like this
```
16 [B]
32 [B]
...
1 [kB]
2 [kB]
...
1 [MB]
2 [MB]
...
```

What do you think?

File: iceoryx_examples/iceperf/iceoryx_wait.hpp
Comment (elBoberido) [inline iceoryx_examples/iceperf/iceoryx_wait.hpp:1]: Please also add your copyright

File: iceoryx_examples/iceperf/iceoryx_wait.cpp
Comment (elBoberido) [inline iceoryx_examples/iceperf/iceoryx_wait.cpp:35]: This is superfluous with the `WaitSet` API

**Review by elBoberido — COMMENTED** [review]
Nice. Just a few small remarks. Looks good overall.

File: iceoryx_examples/iceperf/iceoryx.cpp
Comment (anderay) [reply iceoryx_examples/iceperf/iceoryx.cpp:29]: /// @brief construction of the capro service description using fixed strings to create an event service description
    ServiceDescription(const IdString_t& service,
                       const IdString_t& instance,
                       **const IdString_t& event**,
                       ClassHash m_classHash = {0U, 0U, 0U, 0U},
                       Interfaces interfaceSource = Interfaces::INTERNAL) noexcept;

File: iceoryx_examples/iceperf/iceperf_leader.cpp
Comment (anderay) [reply iceoryx_examples/iceperf/iceperf_leader.cpp:101]: Yes, would be nice ;)

File: iceoryx_examples/iceperf/iceoryx_wait.cpp
Comment (anderay) [reply iceoryx_examples/iceperf/iceoryx_wait.cpp:1]: I hereby resign from my rights to the code.

Comment (elBoberido) [reply iceoryx_examples/iceperf/iceoryx_wait.cpp:1]: Okay, in this case we have to check whether a generic `Copyright 2023, Eclipse Foundation and the iceoryx contributors. All rights reserved.` would be sufficient. We did not yet have the situation of new files and no copyright holder.

Btw, you need to sign the ECA else we cannot merge the PR.

Fun fact, according to the law you cannot really resign from copyright rights. You will always be the copyright holder.

cc @budrus @mossmaurice

Comment (elBoberido) [reply iceoryx_examples/iceperf/iceoryx_wait.cpp:3]: @anderay it seems to be okay to resign from the copyright but you need to sign the ECA. This is required by the Eclipse Foundation and we cannot circumvent this since it is also part of the CI check.

**Comment by elBoberido** [comment]
@anderay do you plan to continue working on this PR?

**Comment by andrzejpolanskiaptiv** [comment]
> @anderay do you plan to continue working on this PR?

What else is needed?

**Comment by elBoberido** [comment]
@andrzejpolanskiaptiv the biggest issue is the [Eclipse ECA](https://www.eclipse.org/legal/ECA.php). Since we are an Eclipse project, all contributors must sign the ECA. There is also a CI check for this which prevents merging.

If you want, you can also implement the proposal from https://github.com/eclipse-iceoryx/iceoryx/pull/1860#discussion_r1097945811 but that is not required. Your answer was a bit ambiguous

**Comment by andrzejpolanskiaptiv** [comment]
"You successfully submitted an Eclipse Contributor Agreement"
But seems not be visible here...

**Comment by andrzejpolanskiaptiv** [comment]
> @andrzejpolanskiaptiv It might take some time until everything is synchronized and the check passes. Just in case, did you sign with the same e-mail address as you used for your commits?

I did it long time ago. Now I see that in commits my company email is stated whilst in ECA private was used. Perhaps that was the problem. I have changed address in Eclipse account, will see...

**Review by elBoberido — COMMENTED** [review]
Looks good. Just a few minor things.
- please tick the checkboxes from the PR author checklist
- please add `iox-#2003 ` to your commit messages, e.g. `iox-#2003 Iceperf example - ein bisschen less advertisement` (I created #2003 for this purpose)
- please add an entry to `iceoryx-unreleased.me` like `- Extend 'iceperf' with 'WaitSet' [#2003](https://github.com/eclipse-iceoryx/iceoryx/issues/2003)` in the `Features` section

**Comment by anderay** [comment]
here you are

**Review by elBoberido — COMMENTED** [review]
Okay, we are really really close. Only the bazel build needs some adjustments and then I can approve the PR :)

Could you please add `iceoryx_wait.hpp/cpp` to `BAZEL.build`. Right after `iceoryx_c.hpp/cpp`

**Review by elBoberido — APPROVED** [review]

**Review by mossmaurice — APPROVED** [review]
LGTM, thanks for the PR! :pray:

**Pull request merged**
